internal/cluster/distribution/eks/ekscluster/eks.go

```diff
@@ -38,9 +39,10 @@ type CreateClusterEKS struct {
 	// Subnets for EKS master and worker nodes. All worker nodes will be launched in the same subnet
 	// (the first subnet in the list - which may not coincide with first subnet in the cluster create request payload as
 	// the deserialization may change the order) unless a subnet is specified for the workers that belong to a node pool at node pool level.
-	Subnets  []*Subnet  `json:"subnets,omitempty" yaml:"subnets,omitempty"`
-	IAM      ClusterIAM `json:"iam,omitempty" yaml:"iam,omitempty"`
-	LogTypes []string   `json:"logTypes,omitempty" yaml:"logTypes,omitempty"`
+	Subnets          []*Subnet               `json:"subnets,omitempty" yaml:"subnets,omitempty"`
+	IAM              ClusterIAM              `json:"iam,omitempty" yaml:"iam,omitempty"`
+	EncryptionConfig []*eks.EncryptionConfig `json:"encryptionConfig,omitempty" yaml:"encryptionConfig,omitempty"`
```

Comment:
Does it have to be a pointer? (TBH I have no idea why Subnet is a pointer).

Reply:
The AWS API expects it to be `[]*eks.EncryptionConfig` (slice of pointers) and I wanted to avoid any unnecessary data transformation logic, so this seemed more straightforward, I don't really see any benefits of changing it to non-pointer and doing the additional data transformation aside from aesthetic gain.

Reply:
[Fixed](https://github.com/banzaicloud/pipeline/compare/6472650b045852246c96e9524c145a422a7049b6..f5bb49ddce4628811c995a6da8cf2d09409a1d0b).